src/Microsoft.Health.Dicom.Core/Features/Retrieve/RetrieveResourceService.cs

```diff
@@ -224,6 +225,7 @@ public async Task<RetrieveResourceResponse> GetInstanceResourceAsync(RetrieveRes
 
         // eagerly doing getFrames to validate frame numbers are valid before returning a response
         Stream stream = await _blobDataStore.GetFileAsync(instance.VersionedInstanceIdentifier.Version, partition, instance.InstanceProperties.FileProperties, cancellationToken);
+        _dicomRequestContextAccessor.RequestContext.BlobBytesEgress = stream?.Length ?? 0;
```

Comment:
For frames I see you are logging for retrieving the file, should we do for the whole file or just for the frames they asked?

Reply:
Seems like we want to bill for the file we are downloading